### PR TITLE
Add prefix blacklist to filter out WebAPI pages

### PR DESCRIPTION
**Pull request by jbzdarkid**
There is no / added on so that we ignore http://wiki.tf/WebAPI as well.

### CODE REVIEW

File: reports.py [hunk 1]
```diff
@@ -78,14 +78,16 @@ def allArticlesEn(self, pagetitle, langs, blacklist, summ):
         def isGoodTitle(title):
             if title in blacklist:
                 return False
-            badguys = [ 'OTFWH', 'titles', 'Archive', 'Header', 'Footer', 'diff' ]
-            badguys.extend(langs)
-            for bad in badguys:
-                if title.endswith('/' + bad):
+            prefixes = [ 'WebAPI' ]
+            for prefix in prefixes:
+                if title.startswith(prefix):
```
Comment (EtiennePerot) [inline reports.py]: `if title.startswith(prefix + '/'):`

Comment (jbzdarkid) [inline reports.py]: [[WebAPI]] is a page that shouldn't be translated.

Comment (EtiennePerot) [inline reports.py]: Similarly to how a "suffix" means "anything slash suffix", I'd expect "prefix" to mean "prefix slash anything". If the `WebAPI` page itself shouldn't be translated, that's just a separate category of exemptions. The code should reflect that.

``` python
pages = [ 'WebAPI ']
prefixes = [ 'WebAPI ']
suffixes = [ 'OTFWH', 'titles', 'Archive', 'Header', 'Footer', 'diff', 'prop' ]
suffixes.extend(langs) # This adds 'es', 'fr', 'de', etc to the suffixes
return not (
  title in pages
  or any(title.startswith(prefix + '/') for prefix in prefixes)
  or any(title.endswith('/' + suffix) for suffix in suffixes)
)
```

**Comment by jbzdarkid** [comment]
Updated with your code.

**Pull request merged**